frontend/src/components/module-viewer/SideBar.tsx

```diff
@@ -72,6 +72,26 @@ const Sidebar = ({
     onClose: onCloseSaveModal,
   } = useDisclosure();
 
+  const handlePageLeave = useCallback((e: BeforeUnloadEvent) => {
+    const confirmationMessage = "Some message";
+    e.preventDefault();
+    e.returnValue = confirmationMessage;
+  }, []);
+
+  const cb = useRef<any>(handlePageLeave);
```

Comment:
Can you fix this? You can probably just use the `Function` type or something, or some callback type.